### PR TITLE
Hide `ordering` a bit

### PR DESCRIPTION
**Pull request by joschmitt**
I renamed the `ordering` keyword in polynomial rings to `internal_ordering` as a first attempt in the direction of https://github.com/oscar-system/Oscar.jl/issues/3040 and a basis for discussion how this should proceed (if at all).

CC @YueRen (because you seemed interested...)

### CODE REVIEW

File: src/MPoly.jl [hunk 1]
```diff
@@ -1211,7 +1211,7 @@ end
 ################################################################################
 
 function _change_mpoly_ring(R, Rx, cached)
-   P, _ = polynomial_ring(R, map(string, symbols(Rx)), ordering = ordering(Rx), cached = cached)
+   P, _ = polynomial_ring(R, map(string, symbols(Rx)), internal_ordering = internal_ordering(Rx), cached = cached)
```
Comment (lgoettgens) [inline src/MPoly.jl:1214]:
```suggestion
   P, _ = polynomial_ring(R, map(string, symbols(Rx)); internal_ordering = internal_ordering(Rx), cached = cached)
```

**Review by lgoettgens — COMMENTED** [review]
I like the general idea of this. some comments

**Comment by YueRen** [comment]
I also like the idea of differentiating between Nemo and Singular monomial orderings by using different names!

**Comment by joschmitt** [comment]
This is the way to go as decided in today's "Kaiserslautern meeting".

**Comment by joschmitt** [comment]
The failing test is "expected", right?

**Comment by joschmitt** [comment]
Please merge this whenever we are ready for the next cascade of breaking releases.

**Comment by joschmitt** [comment]
I removed the deprecations now. If we keep them in, we would also deprecate `ordering` for the Singular.jl rings because they are `<: MPolyRing`. Please tell me, if I am missing something.

**Comment by joschmitt** [comment]
The auto-merge will not work. At least, the "matching" checks haven't started for days now.

**Pull request merged**